File: practica1/python/ejercicio3.py
class Estadisticas:
    def __init__(self, datos):
        self.datos = datos

    def promedio(self):
        suma = 0
        for x in self.datos:
            suma += x
        return suma / len(self.datos)

    def desviacion(self):
        prom = self.promedio()
        suma = 0
        for x in self.datos:
            suma += (x - prom) ** 2
        return (suma / (len(self.datos) - 1)) ** 0.5


# Ejemplo con datos fijos
valores = [1.9, 2.5, 3.7, 2, 1, 6, 3, 4, 5, 2]

est = Estadisticas(valores)

print("El promedio es", est.promedio())
print("La desviacion estandar es", est.desviacion())

File: practica1/java/ejercicio3.java
class Estadisticas {
    private double[] datos;

    public Estadisticas(double[] datos) {
        this.datos = datos;
    }

    public double promedio() {
        double suma = 0;
        for (int i = 0; i < datos.length; i++) {
            suma += datos[i];
        }
        return suma / datos.length;
    }

    public double desviacion() {
        double prom = promedio();
        double suma = 0;
        for (int i = 0; i < datos.length; i++) {
            suma += Math.pow(datos[i] - prom, 2);
        }
        return Math.sqrt(suma / (datos.length - 1));
    }
}

public class Test {
    public static void main(String[] args) {
        // Versión básica con datos fijos
        double[] valores = {1.9, 2.5, 3.7, 2, 1, 6, 3, 4, 5, 2};

        Estadisticas est = new Estadisticas(valores);

        System.out.println("El promedio es " + est.promedio());
        System.out.println("La desviacion estandar es " + est.desviacion());
    }
}